packages/@webex/plugin-meetings/src/statsAnalyzer/index.ts

```diff
@@ -237,82 +182,125 @@ export class StatsAnalyzer extends EventsScope {
+          statsResults: this.statsResults,
+          lastMqaDataSent: this.lastMqaDataSent,
+          mediaType,
+        });
+        newMqa.videoTransmit.push(videoSender);
+      } else if (mediaType.includes('video-recv')) {
+        const videoReceiver = cloneDeep(emptyVideoReceive);
+
+        getVideoReceiverMqa({
+          videoReceiver,
+          statsResults: this.statsResults,
+          lastMqaDataSent: this.lastMqaDataSent,
+          mediaType,
+        });
+        newMqa.videoReceive.push(videoReceiver);
+      } else if (mediaType.includes('audio-share-send')) {
+        const audioSender = cloneDeep(emptyAudioTransmit);
+
+        getAudioSenderMqa({
+          audioSender,
+          statsResults: this.statsResults,
+          lastMqaDataSent: this.lastMqaDataSent,
+          mediaType,
+        });
+        newMqa.audioTransmit.push(audioSender);
+      } else if (mediaType.includes('audio-share-recv')) {
+        const audioReceiver = cloneDeep(emptyAudioReceive);
+
+        getAudioReceiverMqa({
+          audioReceiver,
+          statsResults: this.statsResults,
+          lastMqaDataSent: this.lastMqaDataSent,
+          mediaType,
+        });
+        newMqa.audioReceive.push(audioReceiver);
+      } else if (mediaType.includes('video-share-send')) {
+        const videoSender = cloneDeep(emptyVideoTransmit);
+
+        getVideoSenderMqa({
+          videoSender,
+          statsResults: this.statsResults,
+          lastMqaDataSent: this.lastMqaDataSent,
+          mediaType,
+        });
+        newMqa.videoTransmit.push(videoSender);
+      } else if (mediaType.includes('video-share-recv')) {
+        const videoReceiver = cloneDeep(emptyVideoReceive);
+
+        getVideoReceiverMqa({
+          videoReceiver,
+          statsResults: this.statsResults,
+          lastMqaDataSent: this.lastMqaDataSent,
+          mediaType,
+        });
+        newMqa.videoReceive.push(videoReceiver);
+      }
     });
 
-    getVideoReceiverMqa({
-      videoReceiver: shareReceiver,
-      statsResults: this.statsResults,
-      lastMqaDataSent: this.lastMqaDataSent,
-      isShareStream: true,
-    });
-    mqaData.intervals[0].intervalMetadata.peerReflexiveIP =
-      this.statsResults.connectionType.local.ipAddress[0];
+    newMqa.intervalMetadata.peerReflexiveIP = this.statsResults.connectionType.local.ipAddress[0];
 
     // Adding peripheral information
-    mqaData.intervals[0].intervalMetadata.peripherals = [];
-    mqaData.intervals[0].intervalMetadata.peripherals.push({
-      information: _UNKNOWN_,
-      name: MEDIA_DEVICES.SPEAKER,
+    newMqa.intervalMetadata.peripherals = [];
+
+    let audioTrackLabel = _UNKNOWN_;
+    let videoTrackLabel = _UNKNOWN_;
+
+    // Find track labels
+    Object.keys(this.statsResults).forEach((mediaType) => {
+      if (mediaType.includes('video-send') && this.statsResults[mediaType].trackLabel) {
+        videoTrackLabel = this.statsResults[mediaType].trackLabel;
```

Comment:
hmm not sure about this... in reality we are only ever going to have a single sender for audio and video. Even WCME is hardcoded for that right now (it doesn't support publishing more than 1 track of same type at the same time). This code here is written in a generic way to support many senders (it iterates over the statsResults keys) but the output is just a single audio/video track label - so it just uses the last one it finds, so it won't work correctly if we do have multiple senders anyway. Maybe we should just make it clear that we only have one sender and not bother with "video-send-${idx}" but just have "video-send" and not have the loop here but just access this.statsResults['video-send'].trackLabel?